src/test/kotlin/com/ampnet/reportservice/service/TemplateDataServiceTest.kt

```diff
@@ -0,0 +1,113 @@
+            Mockito.`when`(walletService.getWalletsByOwner(listOf(userUuid))).thenReturn(listOf(testContext.wallet))
+        }
+        suppose("Blockchain service will return transactions for wallet") {
+            testContext.transactions = listOf(
+                createTransaction(
+                    mintHash, userWalletHash, "1000",
+                    TransactionsResponse.Transaction.Type.DEPOSIT
+                ),
+                createTransaction(
+                    userWalletHash, projectWalletHash, "1000",
+                    TransactionsResponse.Transaction.Type.INVEST
+                ),
+                createTransaction(
+                    projectWalletHash, userWalletHash, "1000",
+                    TransactionsResponse.Transaction.Type.CANCEL_INVESTMENT
+                ),
+                createTransaction(
+                    projectWalletHash, userWalletHash, "10",
+                    TransactionsResponse.Transaction.Type.SHARE_PAYOUT
+                ),
+                createTransaction(
+                    userWalletHash, burnHash, "10",
+                    TransactionsResponse.Transaction.Type.WITHDRAW
+                )
+            )
+            Mockito.`when`(blockchainService.getTransactions(testContext.wallet.hash))
+                .thenReturn(testContext.transactions)
+        }
+        suppose("Blockchain service will return wallets for hashes") {
+            testContext.wallets = listOf(
+                createWalletResponse(UUID.randomUUID(), userUuid, hash = userWalletHash),
+                createWalletResponse(UUID.randomUUID(), projectUuid, hash = projectWalletHash)
+            )
+            Mockito.`when`(walletService.getWalletsByHash(setOf(mintHash, userWalletHash, projectWalletHash, burnHash)))
+                .thenReturn(testContext.wallets)
+        }
+        suppose("User service will return a list of users") {
+            testContext.user = createUserResponse(userUuid)
+            Mockito.`when`(userService.getUsers(setOf(userUuid, projectUuid)))
+                .thenReturn(listOf(testContext.user))
+        }
+        suppose("Project service will return a list of projects") {
+            testContext.project = createProjectsResponse(projectUuid)
+            Mockito.`when`(projectService.getProjects(listOf(userUuid, projectUuid)))
+                .thenReturn(listOf(testContext.project))
+        }
+        verify("Template data service can get user transactions") {
+            val user = testContext.user
+            val project = testContext.project
+            val transactions = templateDataService.getUserTransactionsData(userUuid).transactions
+            assertThat(transactions).hasSize(5)
+
+            val depositTx = transactions.first { it.type == TransactionsResponse.Transaction.Type.DEPOSIT }
+            assertThat(depositTx.from).isEqualTo(platformWalletName)
+            assertThat(depositTx.to).isEqualTo("${user.firstName} ${user.lastName}")
```

Comment:
`"${user.firstName} ${user.lastName}"` this is used multiple times, it would be easier to prvoide function getUsername(user) -> String

Just a reminder for the next time.